### ISSUE TITLE
Back button resets report filters

### ISSUE THREAD

**Issue by jonkri** [mysociety/fixmystreet#1920]
### Description of problem

Pressing the browser's Back button resets report filters.

### Version

Presumably at least 2.0, 2.1, 2.2; confirmed on both FixMyStreet and FixaMinGata.

### Steps to reproduce

1. Search for any location with pins with more than one state (like [here](https://www.fixmystreet.com/around?latitude=51.50642;longitude=-0.12721&js=1))
2. Use the first dropdown to select a filter, for instance, changing from *All* to *Closed*
3. Click on a report
4. Press the browsers "native" Back button (not *Back to all reports*)

### Actual results

The filters are reset. The first dropdown shows *All*.

### Expected results

The same filter configuration as appeared after Step 2 above - that is, the first dropdown showing the value *Closed*.

### How reproducible

Seemingly always, at least using Mozilla Firefox and Internet Explorer 11.

Related to #1649.

**Comment by dracos** [comment]
For me, following your instructions, clicking Back the first time doesn't appear to do anything at all, but note that the URL updates to what it should be, the spinner appears to show it is loading data, but it doesn't show anything or update the sidebar, etc. Clicking Back again then does work but (as you've clicked back to your original page now) shows all pins (so for *that URL* is working). I swear this used to work, but it doesn't appear to have done and can certainly confirm it is not now. Hohum.

It looks like on a report page -> list page back, the JS popState is updating the (hidden) filters and fetching the right data but not actually switching back to showing the list page. `fixmystreet.display.reports_list` is (should be?) idempotent, so I think calling this on any filter change would be okay and fix this. #1921 PR opened.

**Comment by dracos** [comment]
Thanks for letting us know, I've merged a fix for this now.
